### DISCUSSION TITLE
Document key generation when variables `undefined` vs `{}`

### DISCUSSION THREAD

**Discussion by mattfysh** [urql-graphql/urql#1689] [Q&A] [answered]
In my app, I have:

* a sidebar which uses `operationStore(listBoards)`
* a dislocated component which deletes a board, and refreshes the sidebar query via `client.query(listBoards, {}, { requestPolicy: 'network-only' })`
* both components import `listBoards` from the same file where the graphql query text is stored

This did not initially work, because `{}` and `undefined` were not the same for key generation purposes. I was able to fix by using `client.query(listBoards, undefined, { requestPolicy: 'network-only' })`

Currently when a document key hash is generated based on the query and (sorted) variables, it uses the exact value passed into the function. Therefore, the following two psuedo-code examples result in different keys:

1. `hash(myQuery)` ... aka `hash(myQuery, undefined)`
2. `hash(myQuery, {})`

Should this be the case, or should the default value for variables always be `{}`?

**Comment by kitten** [comment, marked as answer]
Hm, yea, I believe we could improve this and this seems accidental
